query/query_manager.rs

```diff
@@ -48,14 +51,35 @@ impl QueryManager {
         source_query: &str,
     ) -> Result<(), Box<QueryError>> {
         event!(Level::TRACE, "Running schema query:\n{}", query);
-        match query {
+        let query_profile = QueryProfile::new(tracing::enabled!(Level::TRACE));
```

Comment:
Validation of new / changed schema constraints gets expensive fast.

Reply:
For sure lol